src/plugins/intel_cpu/src/mkldnn_graph_optimizer.cpp

```diff
@@ -1704,16 +1716,25 @@ void MKLDNNGraphOptimizer::FusePerformedAsScaleShiftAndFakeQuantize(MKLDNNGraph
         }
     };
 
-    auto isSuitableScaleShiftNode = [getConstPort](MKLDNNNodePtr node) {
+    auto getNonConstPort = [](const MKLDNNNodePtr node) -> size_t {
+        std::vector<size_t> nonConstPorts;
+        for (size_t i = 0; i < node->getParentEdges().size(); i++) {
+            const auto& parent = node->getParentEdgeAt(i)->getParent();
+            if (!(parent->getType() == Input && parent->isConstant()))
+                nonConstPorts.push_back(i);
+        }
+        // there are more than 1 nonconst port or missed
+        if (nonConstPorts.size() != 1)
+            return -1;
+        return nonConstPorts[0];
+    };
+
+    auto isSuitableScaleShiftNode = [getNonConstPort](MKLDNNNodePtr node) {
         if (one_of(node->getAlgorithm(), EltwiseAdd, EltwiseSubtract, EltwiseMultiply, EltwiseDivide, EltwiseMulAdd)) {
-            MKLDNNNode *parent = nullptr;
-            if (node->getAlgorithm() != EltwiseMulAdd) {
-                const auto constPort = getConstPort(node);
-                if (constPort == -1) {
-                    return false;
-                }
-                parent = node->getParentEdgesAtPort(1 - constPort)[0]->getParent().get();
-            }
+            const auto nonConstPort = getNonConstPort(node);
```

Comment:
This would required some bool flag (to track if nonConstPort already found) anyway
Also I have removed getConstPort and now getNonConstPort is use in two places in scope of this transformation (See code below)
The best way to handle it would be std::optional which we have no access to at the moment